build.gradle

```
@@ -54,6 +54,11 @@ subprojects { subproject ->
 	apply plugin: 'maven-publish'
 	apply plugin: 'signing'
 
+	java {
+		sourceCompatibility = JavaVersion.VERSION_1_8
+		targetCompatibility = JavaVersion.VERSION_1_8
+    }
```

Sensible addition overall, given the reality that WALA needs more work before it can run under Java 9 and later. Just a minor tweak: according to [the documentation](https://docs.gradle.org/current/dsl/org.gradle.api.plugins.JavaPluginExtension.html#N11163), `targetCompatibility` defaults to `sourceCompatibility`. So let's avoid repeating ourselves and just set the desired version once.

```suggestion
	java.sourceCompatibility = JavaVersion.VERSION_1_8
```

---

That being said, we do currently run a limited subset of tests under Java 11:

https://github.com/wala/WALA/blob/b966a4c812e4dedc6a1dee6bbc2a297f5e3896d5/.travis.yml#L35-L36

https://github.com/wala/WALA/blob/b966a4c812e4dedc6a1dee6bbc2a297f5e3896d5/travis/script-gradle#L16

What happens to those tests if we explicitly set `java.sourceCompatibility` to Java 1.8? Perhaps they continue to run as intended, with the Java 11 compiler operating in some sort of Java 1.8 compatibility mode. I'm not sure. But we should pay close attention to that when reviewing Travis CI job outcomes for this pull request.